java/BUILD

```diff
@@ -16,3 +16,21 @@ docker_build(
         # cmd = ["/path/to/deploy.jar", "--option1", ...]
     ],
 )
+
+# The default jetty tarball has a top-level versioned directory. We need to strip this directory
+# so that we can unpack directly into /jetty in our final image.
+genrule(
+    name = "unpacked_jetty",
+    srcs = ["@jetty//file"],
+    outs = ["_unpacked_jetty.tar.gz"],
+    cmd = "mkdir -p $(@D)/jetty && tar --strip-components=1 -xzf $(<) -C $(@D)/jetty && tar -czf $(@) -C $(@D) jetty  && rm -rf $(@D)/jetty",
```

Comment:
Switched it to something closer to yours. I moved the jetty BUILD file out to it's own file so we can buildifier it.